packages/client/src/components/block-defaults/upload-block/UploadBlock.tsx

```diff
@@ -59,6 +61,18 @@ export const UploadBlock: BlockComponent = observer(({ id }) => {
 
             // save it as the value
             setData('value', fileLocation);
+
+            // extract te file extension from the file name
+            const fileExtension = `.${file.name
+                .split('.')
+                .pop()
+                .toLowerCase()}`;
+
+            // clear the value if fileExtension does not exist
```

Comment:
Can we do this extension check before we upload the file on ln:49

Reply:
moved up the extension check